internal/cluster/cluster.go

```diff
@@ -436,12 +467,16 @@ func (m *Manager) autoAssignMachineNetworkCidr(c *common.Cluster) error {
 	 * when at least one of the VIPs is defined.
 	 * In DHCP mode the aim is to get from DB only clusters that are candidates for machine network CIDR auto assign
 	 * The cluster query is for clusters that have their DHCP mode set (vip_dhcp_allocation), the machine network CIDR empty, and in status insufficient, or pending for input.
+	 * In the SNO case, machine networks are set in case that the machine CIDR is not set yet, and the address families of the service and cluster networks are the same,
+	 * and the available networks to be set as machine network each have single CIDR.
 	 */
 	var err error
 	if swag.BoolValue(c.VipDhcpAllocation) {
 		err = m.tryAssignMachineCidrDHCPMode(c)
 	} else if !swag.BoolValue(c.UserManagedNetworking) {
 		err = m.tryAssignMachineCidrNonDHCPMode(c)
+	} else if swag.StringValue(c.HighAvailabilityMode) == models.ClusterHighAvailabilityModeNone {
```

Comment:
does it consider none platform for multiple hosts? asking in context of https://github.com/openshift/assisted-service/blob/e3ec424115f9758e4094970eab739eeeed2b3de5/api/hiveextension/v1beta1/agentclusterinstall_types.go#L269

Reply:
Yes.  It takes the networks from the inventories of all hosts.
BTW, it is for SNO not none platform (ClusterHighAvailabilityModeNone).